src/context.rs

```diff
@@ -96,72 +136,32 @@ impl ProverContext {
 
 impl Drop for ProverContext {
     fn drop(&mut self) {
+        _strategy_cache_reset();
         unsafe {
-            _setup_cache_reset();
-
-            let cuda_ctx = _CUDA_CONTEXT.take().expect("cuda ctx");
-            cuda_ctx.destroy().expect("destroy cuda ctx");
-
-            _DEVICE_ALLOCATOR
-                .take()
-                .unwrap()
-                .free()
-                .expect("free allocator");
-            _SMALL_DEVICE_ALLOCATOR
-                .take()
-                .unwrap()
-                .free()
-                .expect("free small allocator");
-            _HOST_ALLOCATOR
-                .take()
-                .unwrap()
-                .free()
-                .expect("free allocator");
-            _SMALL_HOST_ALLOCATOR
-                .take()
-                .unwrap()
-                .free()
-                .expect("free small allocator");
-            _EXEC_STREAM
-                .take()
-                .unwrap()
-                .inner
-                .destroy()
-                .expect("destroy stream");
-            _H2D_STREAM
-                .take()
-                .unwrap()
-                .inner
-                .destroy()
-                .expect("destroy h2d stream");
-            _D2H_STREAM
-                .take()
-                .unwrap()
-                .inner
-                .destroy()
-                .expect("destroy d2h stream");
-
-            drop(_STRATEGY_CACHE.take());
+            CONTEXT = None;
         }
     }
 }
 
-static mut _CUDA_CONTEXT: Option<CudaContext> = None;
-static mut _EXEC_STREAM: Option<Stream> = None;
-static mut _H2D_STREAM: Option<Stream> = None;
-static mut _D2H_STREAM: Option<Stream> = None;
+fn get_context() -> &'static ProverContextSingleton {
+    unsafe { CONTEXT.as_ref().expect("prover context") }
+}
+
+fn get_context_mut() -> &'static mut ProverContextSingleton {
+    unsafe { CONTEXT.as_mut().expect("prover context") }
+}
 
 pub(crate) fn get_stream() -> &'static CudaStream {
-    unsafe { &_EXEC_STREAM.as_ref().expect("execution stream").inner }
+    &get_context().exec_stream.inner
 }
 
 pub(crate) fn get_h2d_stream() -> &'static CudaStream {
-    // unsafe { &_H2D_STREAM.as_ref().expect("host to device stream").inner }
+    // &get_context().h2d_stream.inner
```

Comment:
is this deliberate?

Reply:
yes, although the code was commented out, it was valid code that would have compiled before but would not compile now so i replaced it with what would compile now